submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+
+The cable driver configuration in Submariner is relatively limited, as one can only select one driver to use at a time and there’s minimal
+support for customization of the configuration. The goals of this proposal for the extension of the cable driver configuration are:
+
+- Maintain backward compatibility with the current cable driver configuration
+- Create a custom cable driver configuration that allows for the setup of one, many or combined drivers. In addition, the configuration
+should allow for further customizations based on the selected driver(s).
+
+Cable driver configuration parameters fall into two categories based on their scope:
+
+- Gateway Pod Local - these are parameters that hold configuration parameters specific to each Pod (values vary from Pod to Pod).
+- Namespace Local - these are configuration parameters that you would like to share across a namespace (in this case the submariner-operator
+namespace).
+
+The proposed approach for additional submariner cable driver configuration parameters is:
+
+- To continue to use the existing environment variables injected into the Gateway Pod (with the exception of PSKs/secrets). These environment
+variables will come from the `submariner.io_submariners.yaml`.
+- To extend `submariner.io_submariners.yaml` with some new global parameters for cable driver configuration in a cluster set.
+- To use a volume mounted ConfigMap for more detailed cable driver configurations (as these may vary).
+- To use mounted secrets for parameters that need to be protected, e.g. the Pre-Shared Key (PSK) and to abandon the use of environment variables
+for secrets.
+
+It's important to consider the benefit of volume mounted ConfigMaps and Secrets versus their use as environment variables injected into the Pod.
+With volume mounted maps/secrets, updates are automatically reflected in the Pod, so cable driver configurations can be updated and
+acted on without having to restart the Gateway Pod. The cable driver itself may need to be restarted which may have implications on existing/active
+traffic flows. The ConfigMap can be marked as `optional` to avoid blocking Pod startup. Environment variables are not updated after a ConfigMap
+or Secret update; The whole Pod will need to be restarted to deal with any configuration changes to environment variables.
+
+> **_NOTE:_** As secrets are used in Submariner it's also critical to ensure the Secrets are encrypted in ETCD.
+
+#### Multiple cable driver configuration
+
+The proposed global configuration parameter extensions to `submariner.io_submariners.yaml` to configure multiple cable drivers are listed
+in the following table:
+
+<!-- markdownlint-disable line-length -->
+| Parameter name              | Description
+|:----------------------------|:----------------------------------------------------------------------------|
+| cableDrivers                | An array of strings listing the drivers to configure as separate cables.
```

Comment:
Global config should remain a single driver, policies are an optional addition on top.

Reply:
The point is that you can connect with different cables to different clusters. Not everything is only vxlan OR libreswan tunnels...

Reply:
That's not my point, rather that the default config should be global and you could optionally specify policies on top to have a different connectivity type for specific areas in the whole clusterset

Reply:
These params are extensions to the existing global configuration for 1 or more cable driver(s)... 

So the single cable driver configuration that exists in `submariner.io_submariners.yaml`  would still exist and that cable driver would be assumed to be the default as specified on L401.